### ISSUE TITLE
sddm + wayland displayserver + wayland session = stdout/err gets passed along

### ISSUE THREAD

**Issue by hsitter** [sddm/sddm#1419]
On the develop branch when using wayland as display server (weston) and logging into a wayland session (plasma) the stdout/stderr [from systemd](https://www.freedesktop.org/software/systemd/man/systemd-journald.service.html#Stream%20logging) appear to get passed along such that qt applications started directly from the session start script log to **sddm's** stderr (even with qt built to log to journal because of [stderr detection logic](https://invent.kde.org/qt/qt/qtbase/-/blob/a73ec95c1e5aea907ac4618122f4715924c883e6/src/corelib/global/qlogging.cpp#L253)). This then results in journald recording output that is clearly not from sddm as being from sddm (because of the limitation documented in the journald page I linked). e.g.  output from kwin but logged as sddm

```
Jun 18 20:30:42 ajax sddm[27108]: kwin_screencast: state changed unconnected  ->  connecting
```

This also appears to fly in the face of the SessionLogFile= setting which says it default logs to .local/share/sddm/wayland-session.log, but that is clearly not true if the output is sent to stout/err. I'm guessing if file logging was actually set up properly we'd then also fail the aforementioned stderr detection in qtbase and consequently bypass the log file and log to journal directly. Hope that makes sense.

TLDR: qdebugs from session processes appear as sddm in the journal. should appear as themselves.

sddm.conf:

```
cat /etc/sddm.conf
[General]
DisplayServer=wayland
EnableHiDPI=true
```

raw journal data showing identifier as sddm

```
Fri 2021-06-18 20:30:42.328826 CEST [s=84977a16168943d09a0892255a740f4c;i=26c3c;b=1f75b620cd094c9187e7f68e61cd159e;m=1a66f710e;t=5c50e856dbcfa;x=4229da6593a51aa2]
    _UID=1000
    _GID=1000
    _CAP_EFFECTIVE=0
    _SELINUX_CONTEXT=unconfined
    _AUDIT_LOGINUID=1000
    _SYSTEMD_OWNER_UID=1000
    _SYSTEMD_SLICE=user-1000.slice
    _SYSTEMD_USER_SLICE=-.slice
    _MACHINE_ID=8bfe3165c83143e4bae58284470f7c1c
    _HOSTNAME=ajax
    PRIORITY=6
    SYSLOG_FACILITY=3
    _TRANSPORT=stdout
    SYSLOG_IDENTIFIER=sddm
    _COMM=kwin_wayland
    _EXE=/usr/bin/kwin_wayland
    _CMDLINE=kwin_wayland --wayland_fd 4 --xwayland /usr/lib/x86_64-linux-gnu/libexec/startplasma-waylandsession
    MESSAGE=kwin_screencast: state changed unconnected  ->  connecting
    _BOOT_ID=1f75b620cd094c9187e7f68e61cd159e
    _AUDIT_SESSION=51
    _SYSTEMD_CGROUP=/user.slice/user-1000.slice/session-51.scope
    _SYSTEMD_SESSION=51
    _SYSTEMD_UNIT=session-51.scope
    _SYSTEMD_INVOCATION_ID=6201a97bdb344536a7be9918e9711851
    _STREAM_ID=755eeb712b1e4717b93e4142d8755d82
    _PID=27108
```

compared to this raw entry written by qdebug directly. we can tell it's from qdebgu because it has additional metadata such as QT_CATEGORY:

```
Fri 2021-06-18 20:11:45.561241 CEST [s=84977a16168943d09a0892255a740f4c;i=26965;b=1f75b620cd094c9187e7f68e61cd159e;m=162adbeb5;t=5c50e41ac0aa1;x=b1bf801ac7fcc890]
    CODE_FUNC=unknown
    SYSLOG_IDENTIFIER=plasmashell
    _TRANSPORT=journal
    _UID=1000
    _GID=1000
    _COMM=plasmashell
    _EXE=/usr/bin/plasmashell
    _CMDLINE=/usr/bin/plasmashell
    _CAP_EFFECTIVE=0
    _SELINUX_CONTEXT=unconfined
    _AUDIT_LOGINUID=1000
    _SYSTEMD_OWNER_UID=1000
    _SYSTEMD_SLICE=user-1000.slice
    _SYSTEMD_USER_SLICE=-.slice
    _MACHINE_ID=8bfe3165c83143e4bae58284470f7c1c
    _HOSTNAME=ajax
    PRIORITY=4
    QT_CATEGORY=default
    CODE_LINE=306
    CODE_FILE=file:///usr/share/plasma/plasmoids/org.kde.plasma.mediacontroller/contents/ui/main.qml
    MESSAGE=file:///usr/share/plasma/plasmoids/org.kde.plasma.mediacontroller/contents/ui/main.qml:306: TypeError: Value is undefined and could not be converted to an object
    _BOOT_ID=1f75b620cd094c9187e7f68e61cd159e
    _AUDIT_SESSION=51
    _SYSTEMD_CGROUP=/user.slice/user-1000.slice/session-51.scope
    _SYSTEMD_SESSION=51
    _SYSTEMD_UNIT=session-51.scope
    _SYSTEMD_INVOCATION_ID=6201a97bdb344536a7be9918e9711851
    _PID=27259
    _SOURCE_REALTIME_TIMESTAMP=1624039905561241
```
